crates/core/src/db/datastore/system_tables.rs

```diff
@@ -1242,11 +1329,22 @@ impl TryFrom<RowRef<'_>> for StScheduledRow<Box<str>> {
     }
 }
 
+/*
 impl From<StScheduledRow<Box<str>>> for ProductValue {
     fn from(x: StScheduledRow<Box<str>>) -> Self {
         product![x.table_id, x.reducer_name]
     }
 }
+*/
+
+thread_local! {
+    static READ_BUF: RefCell<Vec<u8>> = RefCell::new(Vec::new());
+}
+
+/// Read a value from a system table via BSatn.
+fn read_via_bsatn<T: DeserializeOwned>(row: RowRef<'_>) -> Result<T, DBError> {
+    READ_BUF.with_borrow_mut(|buf| Ok(row.read_via_bsatn::<T>(buf)?))
+}
```

Comment:
This is now how all system table rows are going to be read. I haven't finished updating this file yet.